kpi/views/v2/data.py

```diff
@@ -336,17 +338,77 @@ def bulk(self, request, *args, **kwargs):
 
         bulk_actions_validator = DataBulkActionsValidator(**kwargs)
         bulk_actions_validator.is_valid(raise_exception=True)
+        audit_logs = []
+        if request.method == 'DELETE':
+            # Prepare audit log
+            data = copy.deepcopy(bulk_actions_validator.data)
+            # Retrieve all submissions matching `submission_ids` or `query`.
+            # If user is not allowed to see some the submissions (i.e.: user with
+            # partial permissions), the request will be rejected
+            # (aka `PermissionDenied`) before AuditLog objects are saved in DB.
+            submissions = deployment.get_submissions(
+                user=request.user,
+                submission_ids=data['submission_ids'],
+                query=data['query'],
+                fields=['_id', '_uuid']
+            )
+            (
+                app_label,
+                model_name,
+            ) = deployment.submission_model.get_app_label_and_model_name()
+            for submission in submissions:
+                audit_logs.append(AuditLog(
+                    app_label=app_label,
+                    model_name=model_name,
+                    object_id=submission['_id'],
+                    user=request.user,
+                    metadata={
+                        'asset_uid': self.asset.uid,
+                        'uuid': submission['_uuid'],
+                    }
+                ))
+
+        # Send request to KC
         json_response = action_(bulk_actions_validator.data, request.user)
 
+        # If requests has succeeded, let's log deletions (if any)
+        if json_response['status'] == status.HTTP_200_OK and audit_logs:
+            AuditLog.objects.bulk_create(audit_logs)
```

Comment:
Let's say this is Good Enough until we can finally make kobocat a subordinate Django application within KPI :ok_hand: